src/primitives/triangle/scanline_intersections.rs

```diff
@@ -0,0 +1,230 @@
+//! Triangle scanline intersections iterator.
+
+use crate::{
+    geometry::Point,
+    primitives::{
+        line::StrokeOffset,
+        line_join::LineJoin,
+        polyline::scanline_intersections::{extend, touches},
+        thick_segment::ThickSegment,
+        Line, Triangle,
+    },
+};
+
+/// Type of scanline.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub enum PointType {
+    /// Represents part of the stroke.
+    Border,
+
+    /// Represents the interior of the shape.
+    Inside,
```

Comment:
Couldn't these be named `Stroke` and `Fill`? We also use the same naming convention for `stroke_area` and `fill_area`. Or do these variants have a different meaning?

Reply:
Ah, no these are just a holdover from the previous triangle implementation. They can changed.